core/src/test/java/org/elasticsearch/search/aggregations/InternalAggregationTestCase.java

```diff
@@ -122,4 +135,56 @@ protected final T createUnmappedInstance(String name) {
     protected NamedWriteableRegistry getNamedWriteableRegistry() {
         return namedWriteableRegistry;
     }
+
+    protected ContextParser<Object, Aggregation> instanceParser() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
```

Comment:
NamedWriteableRegistry and NamedXContentRegistry are two different beasts, so maybe you're suggestion to have an abstract method like getNamedXContentRegistry in each test?